Event Streams Working Draft 2 (#9)
oasis-tcs/amqp-specs: event-streams/src/event-streams-v1.0-wd01.md

attach a link, but rather point to a different AMQP container where the
requested or assigned partition is available.

Partition-bound links provide the consumer with the assurance that all

> transfers belong to the given partition, which is desireable when the

Reviewer: misspelt word
```suggestion
transfers belong to the given partition, which is desirable when the
```

partition, the event log node MAY bind a partition to the link as described in
[4.3](#43-binding-consumer-links).

If the event log node manages, for instance, 10 partitions and there are 12
candidate consumers trying to establish links, one of two scenarios MAY apply:

1) Two of those consumers MAY see their attach requests immediately rejected
   with a `detach-forced` error since there are no partitions left to assign.
2) The event log node MAY instead choose not to deny attaching the link
   outright, but keep further incoming attach requests pending and complete
   those only once a partition becomes again available for assignment. Attach
   requests might therefore remain pending for a substantial time span, whose
   limits MAY be configurable in an implementation specific way.

If the event log node wants to force renegotiation of partition bindings amongst
a consumer group, it SHOULD gracefully close all partition-bound links and the
respective consumers SHOULD attempt to establish a new link, again with

> indicating the consumer group, but without specifying a partition.

Reviewer: ```suggestion
the consumer group, but without specifying a partition.
```

A consumer MUST strip the annotation if it forwards the message onwards.

#### 4.6.4 event-streams-associated-partitions-changed delivery annotation

The `event-streams-assigned-partitions-changed` delivery annotation is an array
of `symbol` values and contains the list of partitions associated with the link
that the transfer is being carried over.

For all partition-agnostic consumer links, the event log node SHOULD add this

> annotation to the next transfer to the consumer once the association of
> partitions with the link has changed.

Reviewer: maybe?
```suggestion
annotation to the next transfer to the consumer every time the association of
partitions with the link changes.
```
